scriptmodules/system.sh

```diff
@@ -369,6 +369,122 @@ function get_rpi_video() {
     export PKG_CONFIG_PATH="$pkgconfig"
 }
 
+function is_multi_user_target() {
+    [[ "$(systemctl get-default)" == "multi-user.target" ]] && return 0
+    return 1
+}
+
+function is_graphical_target() {
+    [[ "$(systemctl get-default)" == "graphical.target" ]] && return 0
+    return 1
+}
+
+function has_video_output_device() {
+    # check if there is a video output device with a connected display
+    if [[ -d /sys/class/drm ]]; then
+        for d in /sys/class/drm/*/ ;
+        do
+            [[ -f "$d/status" ]] && [[ $(cat "$d/status") == "connected" ]] && return 0
+        done
+    fi
+    return 1
+}
+
+function has_render_device() {
+    # check if there is a render note
+    [[ ! -z "$(ls /dev/dri/render*)" ]] && return 0
+    return 1
+}
+
+function get_graphics_platform() {
+    if is_multi_user_target; then
+        # for kmsdrm we need a card for video output and a render device for gl/gles acceleration
+        has_video_output_device && has_render_device && __platform_flags+=(kms)
+        has_video_output_device && __platform_flags+=(kms softpipe)
+    elif is_graphical_target; then
+        __platform_flags+=(x11)
+        # Only allow wayland if drm-backend.so can be used. Video output and render devices are mandatory.
+        has_video_output_device && has_render_device && __platform_flags+=(wayland)
+    else
+        echo "Unknown system target. Please use multi-user.target or graphical target."
+        exit
+    fi
```

Comment:
I removed this message. I only know multi-user.target and graphical.target. I also removed isGraphicalTarget() and isMultiUserTarget() functions because they were only needed in get_graphics_platform().